runtime/artifact.go

```diff
@@ -49,17 +51,16 @@ func (context *TaskContext) UploadS3Artifact(artifact S3Artifact) error {
 		StorageType: "s3",
 	})
 	if err != nil {
-		return err
+		panic(errors.Wrap(err, "failed to Marshal json that should have worked"))
```

Comment:
By "all input is evil" principle, we should not panic if we fail to process data coming from an external entity. We should log the errors message and fail the task nicely, or it may make difficult to diagnose what's going on if things start to explode in production. That said, do these failures go to sentry? If so, I think we are safe to panic.

Reply:
>By "all input is evil" principle, we should not panic if we fail to process data coming from an external entity. 

I mostly agree...

This isn't external...  If the type `queue.S3ArtifactRequest{...}` doesn't work with `json.Marshal(` then that's either always the case or never the case... IMO we could ignore the error because we know it either always happens or never happens... A panic is just a neat assertion here...

Reply:
Basically, I'm trying to say that if an error can never, then we should never propagate it...

I don't see anyway that external input could trigger an error here. So if an error happens we should panic.